### DISCUSSION TITLE
denols LSP configuration fails when package.json exists in parent directories

### DISCUSSION THREAD

**Discussion by MusheghMov** [LazyVim/LazyVim#4800] [Q&A]
Hi,
I encountered an issue while trying Deno for the first time. The denols language server wasn't loading; instead, vtsls was being activated:
<img width="908" alt="image" src="https://github.com/user-attachments/assets/6658d178-17d3-48d0-98c0-1738ee77958e">
I discovered that this was happening because I had a package.json file in my root directory, which caused both denols and vtsls to start simultaneously. While deleting that file resolved the immediate issue, I found that having any of these configuration files in any parent directory of the current project prevents the correct configuration from working:
<img width="711" alt="image" src="https://github.com/user-attachments/assets/884a3bef-0a16-4171-aed9-8a51d253da13">
Here's the relevant configuration code from nvim-lspconfig for reference:
```lua
    if LazyVim.lsp.is_enabled("denols") and LazyVim.lsp.is_enabled("vtsls") then
      local is_deno = require("lspconfig.util").root_pattern("deno.json", "deno.jsonc")
      LazyVim.lsp.disable("vtsls", is_deno)
      LazyVim.lsp.disable("denols", function(root_dir, config)
        if not is_deno(root_dir) then
          config.settings.deno.enable = false
        end
        return false
      end)
    end
```
Is this a known issue, or could there be something wrong with my local configuration?

**Comment by dpetka2001** [comment]
https://github.com/LazyVim/LazyVim/issues/4319#issuecomment-2371981600
